.github/workflows/cross-bootstrap-tools.yml

```diff
@@ -51,17 +60,35 @@ jobs:
           brew install ${{ matrix.pkgs }} || true
       - name: create environment
         run: |
-          echo "GITHUB_WORKSPACE = $GITHUB_WORKSPACE"
-          if [ -n "${{ matrix.cross-bindir }}" ]; then
-            echo "EXTRA_BUILD_ARGS=--cross-bindir=${{ matrix.cross-bindir }}" >> $GITHUB_ENV
-          fi
+          echo "CC=${{ matrix.llvm-bindir }}/clang" >> $GITHUB_ENV
+          echo "CXX=${{ matrix.llvm-bindir }}/clang++" >> $GITHUB_ENV
+          echo "CPP=${{ matrix.llvm-bindir }}/clang-cpp" >> $GITHUB_ENV
+
+          echo "XCC=${{ matrix.llvm-bindir }}/clang" >> $GITHUB_ENV
+          echo "XCXX=${{ matrix.llvm-bindir }}/clang++" >> $GITHUB_ENV
+          echo "XCPP=${{ matrix.llvm-bindir }}/clang-cpp" >> $GITHUB_ENV
+          echo "XLD=${{ matrix.llvm-ld }}" >> $GITHUB_ENV
+
           mkdir -p ../build
           echo "MAKEOBJDIRPREFIX=${PWD%/*}/build" >> $GITHUB_ENV
-          # heh, works on Linux/BSD/macOS ...
           echo "NPROC=`getconf _NPROCESSORS_ONLN 2>/dev/null || getconf NPROCESSORS_ONLN 2>/dev/null || echo 1`" >> $GITHUB_ENV
-      - name: bootstrap bmake
-        run: ./tools/build/make.py --debug $EXTRA_BUILD_ARGS TARGET=${{ matrix.target }} TARGET_ARCH=${{ matrix.target_arch }} -n
+
+      - name: show environment
+        run: |
+          echo Running on $(uname -a)
+          echo "CWD = $PWD"
+          echo "GITHUB_WORKSPACE = $GITHUB_WORKSPACE"
+          echo "Using bmake as ${{ matrix.bmake }}"
+          echo Using CC=${CC}
+          echo Using LD=${LD}
+          echo Using XCC=${XCC}
+          echo Using XLD=${XLD}
+          echo Using MAKEOBJDIRPREFIX=${MAKEOBJDIRPREFIX}
+          echo Using NPROC=${NPROC}
+          echo Using EXTRA_MAKE_ARGS=${EXTRA_MAKE_ARGS}
+
       - name: make kernel-toolchain
-        run: ./tools/build/make.py --debug $EXTRA_BUILD_ARGS TARGET=${{ matrix.target }} TARGET_ARCH=${{ matrix.target_arch }} kernel-toolchain -s -j$NPROC -DWITH_DISK_IMAGE_TOOLS_BOOTSTRAP
+        run: ${{ matrix.bmake }} TARGET=${{ matrix.target }} TARGET_ARCH=${{ matrix.target_arch }} -s -j$NPROC kernel-toolchain -DWITH_DISK_IMAGE_TOOLS_BOOTSTRAP
+
       - name: make buildkernel
-        run: ./tools/build/make.py --debug $EXTRA_BUILD_ARGS TARGET=${{ matrix.target }} TARGET_ARCH=${{ matrix.target_arch }} KERNCONF=GENERIC NO_MODULES=yes buildkernel -s -j$NPROC $EXTRA_MAKE_ARGS
+        run: ${{ matrix.bmake }} TARGET=${{ matrix.target }} TARGET_ARCH=${{ matrix.target_arch }} -s -j$NPROC buildkernel KERNCONF=GENERIC NO_MODULES=yes
```

Comment:
Why elinminate EXTRA_MAKE_ARGS here?

Reply:
```EXTRA_MAKE_ARGS``` was referenced in the original script, it was never set to any value, it could not be set in the call environment. I have left the ```echo``` and would remove that in a subsequent change, could be done right away.
I found it confusing to have to find what ```EXTRA_MAKE_ARGS``` _in this context_ is actually used for, hence I removed the reference.